### PR TITLE
[RFC] fix: flaky 'jobstop() kills entire process tree #6530'

### LINKED ISSUE #10762: flaky test: test/functional\core\job_spec.lua @ 765: jobs jobstop() kills entire process tree
Failed on AppVeyor:

```
[  ERROR   ] 1 error, listed below:
[  ERROR   ] test/functional\core\job_spec.lua @ 765: jobs jobstop() kills entire process tree #6530
test\functional\helpers.lua:403: 
retry() attempts: 425
test/functional\core\job_spec.lua:780: Expected objects to be the same.
Passed in:
(number) 5
Expected:
(number) 4
```
Log: https://ci.appveyor.com/project/neovim/neovim/builds/26670245/job/cxllu8jjm3dagww9#L10724

Code: https://github.com/neovim/neovim/blob/e8bf952f94b843c67bcb14651cd7b997d93f7cff/test/functional/core/job_spec.lua#L771-L780

### PR DESCRIPTION
**Pull request by erw7**
Debug flaky 'jobstop() kills entire process tree #6530'.

fix https://github.com/neovim/neovim/issues/10762

### CODE REVIEW

File: ci/build.ps1 [hunk 1]
```diff
@@ -115,24 +115,25 @@ cmake --build . --config $cmakeBuildType -- $cmakeGeneratorArgs ; exitIfFailed
 
 # Functional tests
 # The $LastExitCode from MSBuild can't be trusted
+$env:TEST_FILE = "test/functional/core/job_spec.lua"
```
Comment (blueyed) [inline ci/build.ps1]: You can also specify `TEST_FILTER` (to not having to comment other tests).
Also `BUSTED_ARGS=--repeat=100` might be good.

**Comment by blueyed** [comment]
Hmm.. pushed 2 commits here, but they do not show up?!

**Comment by blueyed** [comment]
@erw7 
Triggered after 65 attempts: https://ci.appveyor.com/project/neovim/neovim/builds/26727966/job/3jf5ttm0ulw5x971?fullLog=true#L8040

**Comment by blueyed** [comment]
`vctip.exe` showing up causes this: https://ci.appveyor.com/project/neovim/neovim/builds/26727966/job/3jf5ttm0ulw5x971?fullLog=true#L7648

**Comment by blueyed** [comment]
I would not expect it to be a child process, is it?

Or is `os_proc_children` not working correctly?  https://github.com/erw7/neovim/blob/27565ac6f4944e7ad705243453b7d66aaad4405e/src/nvim/os/process.c#L120-L140

**Comment by blueyed** [comment]
Also `conhost.exe` being included as a child proc already (causing the different assertion in the first place) seems incorrect to me.

**Comment by justinmk** [comment]
Nice catch! `vctip.exe` is "Microsoft VC compiler and tools experience improvement data uploader". Sounds like a random clippy-thing.

> I would not expect it to be a child process, is it?
> 
> Or is `os_proc_children` not working correctly?

`tasklist` shows it for that ppid (parent PID), right? 

If Windows/MSVC is starting random child processes, I think it's fine to just allow 0-1 extra children in the test case. 

> Also `conhost.exe` being included as a child proc already (causing the different assertion in the first place) seems incorrect to me.

? conhost.exe is the cmd.exe shell, it is started by `jobstart({string})`. I would expect it to be a child of the job.

**Comment by blueyed** [comment]
Ok, conhost is ok then..

Seems weird for a clippy-thing to show up as a child process for the job/cmd.exe, no?

For the test it would make sense to assert ">= 4" processes there then for Windows (this is just for waiting that they are spawned).

**Comment by justinmk** [comment]
> Seems weird for a clippy-thing to show up as a child process for the job/cmd.exe, no?

Definitely. But we know that it's some random Microsoft popup-like thing, so it's probably not worth investigating...
 
> For the test it would make sense to assert ">= 4" processes there then for Windows (this is just for waiting that they are spawned).

A range of "between 4 and 5" should be enough to avoid false failures, and still keep the test rather strict.

**Comment by blueyed** [comment]
Seems to be good: @erw7 please review / squash. 
I think it might be good to have the process tree output somehow still in case of failures, but we can take it as a snippet from this PR here then also maybe still.

**Comment by erw7** [comment]
I want the range to be 3-5(Probably due to different versions of Windows, but in my environment it will be 3. `conhost.exe` cannot be a child process of `cmd.exe`. It is a child process of `csrss.exe`). I think it looks good.

**Pull request merged**